packages/react-dev-utils/FlowTypecheckPlugin.js

```diff
@@ -0,0 +1,220 @@
+    })
+    .map(line => line.replace(/^Error:\s*/, ''))
+    .join('\n');
+}
+
+function getFlowVersion(global) {
+  return exec(global ? 'flow' : flowBinPath, ['version', '--json'])
+    .then(data => JSON.parse(data.toString('utf8')).semver || '0.0.0')
+    .catch(() => null);
+}
+
+class FlowTypecheckPlugin {
+  constructor() {
+    this.shouldRun = false;
+    this.flowStarted = false;
+    this.flowStarting = null;
+
+    this.flowVersion = require(path.join(
+      __dirname,
+      'package.json'
+    )).dependencies['flow-bin'];
+  }
+
+  startFlow(cwd) {
+    if (this.flowStarted) {
+      return Promise.resolve();
+    }
+    if (this.flowStarting != null) {
+      return this.flowStarting.then(err => {
+        // We need to do it like this because of unhandled rejections
+        // ... basically, we can't actually reject a promise unless someone
+        // has it handled -- which is only the case when we're returned from here
+        if (err != null) {
+          throw err;
+        }
+      });
+    }
+    console.log(chalk.cyan('Starting the flow server ...'));
+    const flowConfigPath = path.join(cwd, '.flowconfig');
+    let delegate;
+    this.flowStarting = new Promise(resolve => {
+      delegate = resolve;
+    });
+    return getFlowVersion(true)
+      .then(globalVersion => {
+        if (globalVersion === null) return;
+        return getFlowVersion(false).then(ourVersion => {
+          if (globalVersion !== ourVersion) {
+            return Promise.reject('__FLOW_VERSION_MISMATCH__');
+          }
+        });
+      })
+      .then(
+        () => new Promise(resolve => {
+          fs.access(flowConfigPath, err => {
+            if (err) {
+              resolve(exec(flowBinPath, ['init'], { cwd }));
+            } else {
+              resolve();
+            }
+          });
+        })
+      )
+      .then(() => exec(flowBinPath, ['stop'], {
+        cwd,
+      }))
+      .then(() => exec(flowBinPath, ['start'], { cwd }).catch(err => {
+        if (
+          typeof err.message === 'string' &&
+          err.message.indexOf('There is already a server running') !== -1
+        ) {
+          return true;
+        } else {
+          throw err;
+        }
+      }))
+      .then(() => {
+        this.flowStarted = true;
+        delegate();
+        this.flowStarting = null;
+      })
+      .catch(err => {
+        delegate(err);
+        this.flowStarting = null;
+        throw err;
+      });
+  }
+
+  apply(compiler) {
+    compiler.plugin('compile', () => {
+      this.shouldRun = false;
+    });
+
+    compiler.plugin('compilation', compilation => {
+      compilation.plugin('normal-module-loader', (loaderContext, module) => {
+        if (
+          this.shouldRun ||
+          module.resource.indexOf('node_modules') !== -1 ||
+          !/[.]js(x)?$/.test(module.resource)
+        ) {
+          return;
+        }
+        const contents = loaderContext.fs.readFileSync(module.resource, 'utf8');
```

Comment:
Is this not slowing down the build? Seems suspicious to read every file if Webpack also does it by itself.